xarray/conventions.py

```diff
@@ -751,6 +751,10 @@ def _encode_coordinates(variables, attributes, non_dim_coord_names):
                 f"'coordinates' found in both attrs and encoding for variable {name!r}."
             )
 
+        # if coordinates set to None, don't write coordinates attribute
+        if "coordinates" in attrs and attrs.get("coordinates") is None:
```

Comment:
```suggestion
        if attrs.get("coordinates") is None or encoding.get("coordinates") is None:
```

We will need to test for `encoding` also

Reply:
@dcherian I think "coordinates" doesn't always exist in attrs and encoding at this point, so `attrs.get("coordinates")` and `encoding.get("coordinates")` might be `None` anyway, and wouldn't be written even without specifying. 

I added in `if "coordinates" in attrs` to check for this, so maybe would need to keep this and do the same for encoding?

Reply:
oh oops that's right. Thanks!